### DISCUSSION TITLE
Does the curl plugin actually pay attention to .curlrc ?

### DISCUSSION THREAD

**Discussion by GK-774** [MusicPlayerDaemon/MPD#1648] [Q&A] [answered]
Does the curl plugin actually read the parameters from /var/lib/mpd/.curlrc and if yes, does it log it´s actions?

While the documentation documentation (https://mpd.readthedocs.io/en/stable/plugins.html#input-plugins) states "... general curl configuration from environment variables such as http_proxy or specified in ~/.curlrc will be in effect." it appears .curlrc is silently ignored.

I have in my .curlrc retry 5 and retry-all-errors to enforce a reconnect when a live stream drops it´s tcp connection (curls default is not reconnect) but apparently no reconnect is ever attempted (and mpd logs it´s well known dreaded "playing silence to avoid xrun" forever, or a stop and start which reconnects immidiately).

Are there any proven workarounds to reconnect to a stream? I´m courious to know how others may have attempted to solve this issue.

**Comment by geneticdrift** [comment, marked as answer]
As far as I can tell `.curlrc` is only read by the curl command line tool but not by `libcurl`.

MPD does set the `.netrc` option to `CURL_NETRC_OPTIONAL`, but it is limited to user name and password:

[CURLOPT_NETRC](https://curl.se/libcurl/c/CURLOPT_NETRC.html)

> Only machine name, user name and password are taken into account (init macros and similar things are not supported). 

https://github.com/MusicPlayerDaemon/MPD/blob/dda0dfc1408e137e11055fec43a7022a1c028aae/src/lib/curl/Setup.cxx#L43

**Reply by MaxKellermann** [reply]
> As far as I can tell `.curlrc` is only read by the curl command line tool but not by `libcurl`.

True, that's a command-line tool feature, not a library feature. The curl documentation sometimes isn't clear enough which is which, and I misunderstood this when I wrote the documentation. It's a MPD documentation bug.

**Reply by jcorporation** [reply]
I created a pull request to fix the documentation bug: #1651

**Comment by GK-774** [comment]
Yes, contrary to what the documentation says, .curlrc is apparently not read by the curl plugin.
